### PR TITLE
Code gen for NEST to support backporting APIs

### PR DESCRIPTION
**Pull request by stevejgordon**
Mostly seems to include format changes which is at least consistent with the other branches.

### CODE REVIEW

**Review by Mpdreamz — APPROVED** [review]
:+1: LGTM

**Comment by stevejgordon** [comment]
Rebased onto master after Transport PR merged. Waiting for CI as a double check before merging.

**Pull request merged**
